neuralprophet/forecaster.py

```diff
@@ -3000,3 +3015,46 @@ def _reshape_raw_predictions_to_forecst_df(self, df, predicted, components):
                         yhat_df = pd.Series(yhat, name=comp).set_axis(df_forecast.index)
                         df_forecast = pd.concat([df_forecast, yhat_df], axis=1, ignore_index=False)
         return df_forecast
+
+    def conformalize(self, df_cal, alpha, method="naive", plotting_backend="default"):
```

Comment:
currently, `predict` implicitly assume that this method has been called beforehand. 
We could:
a) To improve encapsulation we can move the code block from predict here, call this method `conformal_predict` or similar, and have it accept a prediction `df` which is passed on to `predict` internally. 
b) alternatively, this could be split into a three step procedure: `compute_conformalization`, `predict`, `conformalize_prediction`. 

Both would have the advantage of separating regular prediction and conformal prediction.
There may be an even better approach though.

Reply:
Please see PR #1044. This implements what you have for **a)**. There is a `conformal_predict()` method that combines the `conformalize()` and `predict()`, and will need to input most the calibration and test sets for split CP. With this implemented, `conformalize()` will then be removed.

Reply:
Also, for future bootstrapped (e.g. Jackknife+ and CV+ based) CP methods, calibration set will not be needed, which is why `calibration_df` is set to `None` as default in `conformal_predict()`.